app/src/main/java/fr/free/nrw/commons/OkHttpConnectionFactory.java

```diff
@@ -0,0 +1,61 @@
+package fr.free.nrw.commons;
+
+import org.wikipedia.dataclient.SharedPreferenceCookieManager;
+import org.wikipedia.dataclient.okhttp.HttpStatusException;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.annotation.NonNull;
+import okhttp3.Cache;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.logging.HttpLoggingInterceptor;
+
+public final class OkHttpConnectionFactory {
```

Comment:
We do have a provider for getting http client in `NetworkingModule:provideOkHttpClient`. It could be modified to add additional interceptors.

Reply:
Cool, I will de-duplicate things as I go along.